Manage.c

```diff
@@ -1157,15 +1156,17 @@ int Manage_remove(struct supertype *tst, int fd, struct mddev_dev *dv,
 			/* Old kernels rejected this if no personality
 			 * is registered */
 			struct mdinfo *sra = sysfs_read(fd, NULL, GET_DEVS);
-			struct mdinfo *dv = NULL;
+			struct mdinfo *dev = NULL;
+
 			if (sra)
-				dv = sra->devs;
-			for ( ; dv ; dv=dv->next)
-				if (dv->disk.major == (int)major(rdev) &&
-				    dv->disk.minor == (int)minor(rdev))
+				dev = sra->devs;
+			for ( ; dev ; dev=dev->next)
+				if (dev->disk.major == (int)major(rdev) &&
```

Comment:
Please rework this code to add it natural flow:

```diff
diff --git a/Manage.c b/Manage.c
index a0f5b844af3d..a48fe751ba32 100644
--- a/Manage.c
+++ b/Manage.c
@@ -1160,9 +1160,13 @@ int Manage_remove(struct supertype *tst, int fd, struct mddev_dev *dv,
                         * is registered */
                        struct mdinfo *sra = sysfs_read(fd, NULL, GET_DEVS);
                        struct mdinfo *dev = NULL;
-                       if (sra)
-                               dev = sra->devs;
-                       for ( ; dev ; dev=dev->next)
+
+                       if (!sra) {
+                               err = -1;
+                               break;
+                       }
+
+                       for (dev = sra->devs; dev ; dev=dev->next)
                                if (dev->disk.major == (int)major(rdev) &&
                                    dev->disk.minor == (int)minor(rdev))
                                        break;

```

Reply:
fixed.

Reply:
ok, so it must stay same way otherwise we will have to rework more lines (I plan to rework this code soon)